### PR TITLE
Scarlet Monastery: improve Mograine and Withemane encounter

### PR DESCRIPTION
**Pull request by schell244**
## 🍰 Pullrequest
Some improvements for Mograine and Withemane encounter at Scarlet Monastery.
- prevent Mograine from moving while being in fake death state (his corpse was sliding on the ground)
- no longer use `UNIT_FLAG_SPAWNING `in the script
- properly prevent Withemane from getting aggro, before she gets triggered by Mograine
- add sound when Mograine fakes death (source: CMangos https://github.com/cmangos/mangos-classic/blob/dc45231c0b3d852ade709094e3d14f4c9848987a/src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/scarlet_monastery/boss_mograine_and_whitemane.cpp#L187)

### Proof
- https://youtu.be/nPfBZ9lBDzo?t=603

### Issues
- None

### How2Test
- Enter Scarlet Monastery Cathedral
- encounter Mograine and Withemane

### Todo / Checklist
- [X] tested ingame

### CODE REVIEW

File: src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp [hunk 1]
```diff
@@ -84,7 +84,8 @@ struct boss_scarlet_commander_mograineAI : public ScriptedAI
         m_uiHammerOfJustice_Timer = 15000;
 
         //Incase wipe during phase that mograine fake death
-        m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_SPAWNING);
+        m_creature->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_SPAWNING);
```
Comment (ratkosrb) [inline src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp]: Spawning flag should not be used in script at all. Previously it was named "non attackable" so a lot of scripts used it when they shouldn't have. Can you refactor the script to not use it? Combining Immune to Player and NPC together should have the same effect (non attackable by all).

Comment (schell244) [reply src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp]: Ok, I will try.

Comment (schell244) [reply src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp]: I had to add some more changes. The Problem was that Mograine is in combat when he gets the flags Immune to Player and NPC added, and so he would immediately leave fake death and attack player during next `m_creature->SelectHostileTarget() ` check. Now it should work like expected (I did a few tests with bots).

**Comment by schell244** [comment]
Just fixed another edge case. Now PR is ready.

File: src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp [hunk 2]
```diff
@@ -476,14 +508,31 @@ struct boss_high_inquisitor_whitemaneAI : public ScriptedAI
...
-        if (MovementType == POINT_MOTION_TYPE && id == 1)
-            m_creature->SetInCombatWithZone();
-        else
+        if (MovementType == POINT_MOTION_TYPE) 
         {
-            if (m_pInstance)
-            {
-                if (Creature* pMograine = m_pInstance->instance->GetCreature(m_pInstance->GetData64(DATA_MOGRAINE)))
-                    m_creature->SetFacingToObject(pMograine);
+            switch (id) {
```
Comment (ratkosrb) [inline src/scripts/eastern_kingdoms/tirisfal_glades/scarlet_monastery/boss_mograine_and_whitemane.cpp]: Can you please move the opening bracket to a new row.
